nautobot/core/views/renderers.py

```diff
@@ -0,0 +1,217 @@
+                        )
+                        view.queryset = view.queryset.none()
+                table = self.construct_table(view, request=request, permissions=permissions)
+            elif view.action == "destroy":
+                form = form_class(initial=request.GET)
+            elif view.action in ["create", "update"]:
+                initial_data = normalize_querydict(request.GET)
+                form = form_class(instance=instance, initial=initial_data)
+                restrict_form_fields(form, request.user)
+            elif view.action == "bulk_destroy":
+                if request.POST.get("_all"):
+                    if view.filterset_class is not None:
+                        pk_list = [obj.pk for obj in self.filterset_class(request.POST, model.objects.only("pk")).qs]
+                    else:
+                        pk_list = model.objects.values_list("pk", flat=True)
+                else:
+                    pk_list = request.POST.getlist("pk")
+                    initial = {
+                        "pk": pk_list,
+                        "return_url": return_url,
+                    }
+                    form = form_class(initial=initial)
+                table = self.construct_table(view, pk_list=pk_list)
+            elif view.action == "bulk_create":
+                form = view.get_form()
+                if request.data:
+                    table = data.get("table")
+            elif view.action == "bulk_update":
+                if request.POST.get("_all") and view.filterset_class is not None:
+                    pk_list = [obj.pk for obj in view.filterset_class(request.POST, view.queryset.only("pk")).qs]
+                else:
+                    pk_list = request.POST.getlist("pk")
+
+                if "_apply" in request.POST:
+                    form = form_class(model, request.POST)
+                else:
+                    # Include the PK list as initial data for the form
+                    initial_data = {"pk": pk_list}
+                    form = form_class(model, initial=initial_data)
+
+                restrict_form_fields(form, request.user)
+                table = self.construct_table(view, pk_list=pk_list)
+
+        context = {
+            "action_buttons": valid_actions,
+            "changelog_url": changelog_url,
+            "content_type": content_type,
+            "editing": instance.present_in_database,
+            "filter_form": self.get_filter_form(view, request),
+            "form": form,
+            "object": instance,
+            "obj": instance,  # NOTE: This context key is deprecated in favor of `object`.
+            "obj_type": view.queryset.model._meta.verbose_name,  # NOTE: This context key is deprecated in favor of `verbose_name`.
+            "obj_type_plural": view.queryset.model._meta.verbose_name_plural,  # NOTE: This context key is deprecated in favor of `verbose_name_plural`.
```

Comment:
See my comments on the templates above - can we make it so that the templates work with either set of keys, so that we can omit these keys entirely from this new context?